File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length==0)
            return "";
        String prefix = strs[0];

        for (String s : strs) {
            while (s.indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);
                if (prefix == "")
                    return "";
            }
        }

        return prefix;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(!strs.length) return '';

    let prefix = strs[0];
    for(let str of strs){
        while(str.indexOf(prefix)!==0){
            prefix = prefix.slice(0, -1);
            if(prefix === '') return "";
        }
    }
    return prefix
};